extensions/amp-list/0.1/amp-list.js

```diff
@@ -176,12 +185,30 @@ export class AmpList extends AMP.BaseElement {
   }
 
   /** @override */
-  isLayoutSupported(layout) {
-    return isLayoutSizeDefined(layout);
+  isLayoutSupported() {
+    return true;
   }
 
   /** @override */
   buildCallback() {
+    if (this.element.getAttribute('layout') === Layout.CONTAINER) {
+      const doc = this.element.ownerDocument;
+      userAssert(
+        (doc && isAmp4Email(doc)) ||
+          isExperimentOn(this.win, 'amp-list-layout-container'),
+        'Experiment "amp-list-layout-container" is not turned on.'
+      );
```

Comment:
Should we modify `isLayoutSupported` instead? 

We should also only add `supported_layouts: CONTAINER` to the email version of the validator spec.

Reply:
Haven't had an opportunity to look into this more deeply, but it looks like when these assertions in `isLayoutSupported` throw errors the component will still call `buildCallback`. I moved it to `buildCallback` so it would be blocking, but it's not clear to me why this has to be.

Sample with experiment off:
![image](https://user-images.githubusercontent.com/10456171/80021915-43cbf180-84a9-11ea-9b3d-e5df30417fde.png)

Reply:
Right, I guess soft failure mode is precedence here. Is that acceptable?

Reply:
This is OK if the component fails the experiment page as long as it's on a valid AMP page. This is because `layout=container` will _not_ be valid for amp-list for web.

Reply:
Not to belabor this point (since I don't think it'll make much difference in practice), but we're still doing something similar here since we're throwing an exception. Check out the caller of `isLayoutSupported()` vs. `buildCallback()` to see the difference.